src/newsView.js

```diff
@@ -0,0 +1,91 @@
+const NewsApi = require("./newsApi")
+const NewsModel = require("./newsModel")
+
+class NewsView{
+	
+	constructor(model = new NewsModel(), api = new NewsApi()){
+		this.model = model;
+		this.api = api;
+		this.newsBlock = document.querySelector('.news-block');
+		
+
+		(()=>{
+			this.load(()=>{
+				let btn = document.querySelectorAll('.my-btn');
+				this.newsSlot = document.querySelector('#news-slot');
+				window.addEventListener('hashchange', (e) => this.routechange(e))
```

Comment:
Interesting use of hashchange. Not an event listener I was aware of, I'll need to look this up myself. Good that you are using new ideas outside of what the course has directly taught